tests/suites/external_service/external_service_test.go

```diff
@@ -0,0 +1,138 @@
+	Operator  = kudo.Operator{}
+)
+
+var _ = BeforeSuite(func() {
+	Client, _ = client.NewForConfig(KubeConfigPath)
+	kubernetes.CreateNamespace(Client, TestNamespace)
+})
+
+var _ = AfterSuite(func() {
+	Operator.Uninstall()
+	kubernetes.DeleteNamespace(Client, TestNamespace)
+})
+
+//var _ = BeforeSuite(func() {
+//	k8s.Init(KubectlOptions)
+//	kudo.Init(KubectlOptions)
+//	_ = kudo.UninstallOperator(OperatorName, TestNamespace, TestInstance)
+//	_ = k8s.CreateNamespace(TestNamespace)
+//})
+//
+//var _ = AfterSuite(func() {
+//	_ = kudo.UninstallOperator(OperatorName, TestNamespace, TestInstance)
+//	_ = k8s.DeleteNamespace(TestNamespace)
+//	if OperatorVersion != "" {
+//		_, _, err := kudo.OverrideOperatorVersion(OperatorVersion)
+//		if err != nil {
+//			log.Errorf(
+//				"Error reverting operatorVersion from '%s' to '%s': %v",
+//				TestOperatorVersion, OperatorVersion, err,
+//			)
+//		}
+//	}
+//})
+
+func TestService(t *testing.T) {
+	RegisterFailHandler(Fail)
+	junitReporter := reporters.NewJUnitReporter(fmt.Sprintf(
+		"%s-junit.xml", TestName,
+	))
+	RunSpecsWithDefaultAndCustomReporters(t, TestName, []Reporter{junitReporter})
+}
+
+func assertNumberOfCassandraNodes(nodeCount int) {
+	nodes, err := cassandra.Nodes(Client, Operator.Instance)
+	Expect(err).To(BeNil())
+	Expect(len(nodes)).To(Equal(nodeCount))
+}
+
+var _ = Describe("external service", func() {
+
+	It("Installs the operator from the current directory", func() {
+		var err error
+
+		Operator, err = kudo.InstallOperator(OperatorDirectory).
+			WithNamespace(TestNamespace).
+			WithInstance(TestInstance).
+			WithParameters(map[string]string{
+				"NODE_COUNT": strconv.Itoa(NodeCount),
+			}).
+			Do(Client)
+		Expect(err).To(BeNil())
+
+		err = Operator.Instance.WaitForPlanInProgress("deploy")
+		Expect(err).To(BeNil())
+
+		err = Operator.Instance.WaitForPlanComplete("deploy")
+		Expect(err).To(BeNil())
+
+		assertNumberOfCassandraNodes(NodeCount)
+	})
+
+	It("Allows external access to the cassandra cluster", func() {
+		err := Operator.Instance.UpdateParameters(map[string]string{
+			"EXTERNAL_NATIVE_TRANSPORT": "true",
+		})
+		Expect(err).To(BeNil())
+
+		assertNumberOfCassandraNodes(NodeCount)
+
+		log.Infof("Verify that external service is started and has 1 open port")
+		svc, err := kubernetes.GetService(Client, fmt.Sprintf("%s-svc-external", TestInstance), TestNamespace)
+		Expect(err).To(BeNil())
+		Expect(len(svc.Spec.Ports)).To(Equal(1))
+	})
+
+	It("Opens a second port if rpc is enabled", func() {
+		err := Operator.Instance.UpdateParameters(map[string]string{
+			"START_RPC":    "true",
+			"EXTERNAL_RPC": "true",
+		})
+		Expect(err).To(BeNil())
+
+		assertNumberOfCassandraNodes(NodeCount)
+
+		log.Infof("Verify that external service is started and has 2 open ports")
+		svc, err := kubernetes.GetService(Client, fmt.Sprintf("%s-svc-external", TestInstance), TestNamespace)
+		Expect(err).To(BeNil())
+		Expect(len(svc.Spec.Ports)).To(Equal(2))
```

Comment:
Same as above, would be great to check that these ports are in fact the ones that have been configured with name and external port.